### PR TITLE
refactor(hz): hz template

### PR DESCRIPTION
**Pull request by FGYFFFF**
#### What type of PR is this?

refactor
#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
重构 hz 生成的代码的模板

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: refactor hz template for generating code more flexibly
zh(optional): 1. 添加了自定义 package 文件模板功能 2. 添加了覆盖模板机制，写模板的时候可以不用全量写默认模板 3. 增加对同一个文件增量更新的逻辑

### CODE REVIEW

**Review by li-jin-gou — DISMISSED** [review]

File: cmd/hz/util/ast.go [hunk 1]
```diff
@@ -32,18 +32,40 @@ func AddImport(file, alias, impt string) (string, error) {
...
 	err = format.Node(buffer, fset, f)
 	if err != nil {
-		return "", fmt.Errorf("can add import for file: %s", path)
+		return "", fmt.Errorf("can add import for file: %s, err: %v", path, err)
 	}
 
 	return buffer.String(), nil
 }
+
+func AddImportForContent(fileContent []byte, alias, impt string) ([]byte, error) {
```
Comment (welkeyever) [inline cmd/hz/util/ast.go:52]: So many duplicate code with line 30-50, try to do a refactor job?

**Review by Duslia — DISMISSED** [review]

File: cmd/hz/generator/template.go [hunk 2]
```diff
@@ -36,26 +36,40 @@ type TemplateConfig struct {
...
-
-	// custom template files and whether divide method
-	customTpls    map[string]*template.Template
-	pathNameTpls  map[string]*template.Template
-	divideMethods map[string]bool
+	OutputDir    string
+	Config       *TemplateConfig
+	Excludes     []string
+	tpls         map[string]*template.Template // template name -> template
+	tplsInfo     map[string]*Template          // template name -> template info, for getting more template info
```
Comment (welkeyever) [inline cmd/hz/generator/template.go:70]: What is the different between tpls/tplsInfo?
Maybe add more comment to explain？

Comment (FGYFFFF) [reply cmd/hz/generator/template.go]: done~

**Review by Duslia — DISMISSED** [review]

File: cmd/hz/generator/template.go [hunk 3]
```diff
@@ -36,26 +36,40 @@ type TemplateConfig struct {
...
-	dirs      map[string]bool
-
-	// custom template files and whether divide method
-	customTpls    map[string]*template.Template
-	pathNameTpls  map[string]*template.Template
-	divideMethods map[string]bool
+	OutputDir    string
+	Config       *TemplateConfig
+	Excludes     []string
+	tpls         map[string]*template.Template // "template name" -> "Template", it used to get the "parsed template" directly
```
Comment (welkeyever) [inline cmd/hz/generator/template.go:69]: `it used` -> `it is used`？

**Review by wzekin — DISMISSED** [review]

File: cmd/hz/generator/template.go (hunk 3 above)
Comment (li-jin-gou) [reply cmd/hz/generator/template.go]: done

**Review by li-jin-gou — APPROVED** [review]

**Review by welkeyever — APPROVED** [review]

**Review by wzekin — APPROVED** [review]

**Pull request merged**